syntax = "proto2";

package proto;

message AnimationFrame2d {
  // Name of asset file, the png for example this frame comes from.
  optional string asset_name = 1;
  // x, y of frame in texture
  optional float texture_x = 2;
  optional float texture_y = 3;
  // The width / height of the given frame.
  optional float texture_width = 4;
  optional float texture_height = 5;
  // How long to play this animation for.
  optional float duration_sec = 6;
};

message Animation2d {
  // Animations need a unique name, e.g. knight_walk / knight_idle / fire_spirit_attack / etc.
  optional string animation_name = 1;
  // Frames to that compose this animation.
  repeated AnimationFrame2d frames = 2;
};
